### DISCUSSION TITLE
Same left endpoint and left subnet?

### DISCUSSION THREAD

**Discussion by NormandoHall** [strongswan/strongswan#794] [Q&A] [answered]
Is this correct, using the same left endpoint and same left subnet? This is because I have a floating ip (public) assigned to the internal network interface. Or it is better to create a new network interface with another ip?

Thanks

`ipsec.conf`

```
config setup
    # cachecrls=yes
    # charonstart=no
    # strictcrlpolicy=yes
    # uniqueids=no
    # charondebug="dmn 0, mgr 0, ike 1, chd 0, job 0, cfg 1, knl 1, net 1, enc 0, lib 0"

conn %default
    ikelifetime=5m
    lifetime=5m
    margintime=1m
    rekeyfuzz=0%
    keyingtries=30
    reauth=no
    authby=secret
    keyexchange=ike
    mobike=no
    ike=3des-md5-modp1024!
    esp=3des-md5-modp1024-noesn!

conn COMPANY
    left=192.168.10.20
    leftsubnet=192.168.10.20/32
    leftid=192.168.10.20
    leftfirewall=yes
    right=xx.xx.xx.xx
    rightsubnet=172.16.0.0/20
    rightid=xx.xx.xx.xx
    compress=no
    auto=start

```

**Comment by tobiasbrunner** [comment, marked as answer]
> Is this correct, using the same left endpoint and same left subnet?

If that's the IP address you want to use inside the tunnel (i.e. to reach hosts in `172.16.0.0/20`), sure. You don't actually have to configure `leftsubnet` in that case as it will default to `left`, which you also don't have to configure, unless the IKE daemon does not select the correct local IP address automatically.

If you want to use a different address (might also depend on the other end as it has to route traffic from `172.16.0.0/20` back to you), you could also configure any other IP address in `leftsubnet` that's on a local interface (doesn't matter on which, could be on `lo`). So you wouldn't  have to create a virtual interface for that (assuming you are using Linux).

**Reply by NormandoHall** [reply]
Thank you so much @tobiasbrunner for you reply. Yes, runs like a charm. I was tested for two days and it is perfect. Superb software.

Can I asked if my configuration is optimal, specially for `ikelifetime `and `lifetime`? I have not an IPSec deeper knowledge.

We are using **Strongswan V5.7.2** con Centos 7, and the peer is a mobile company that tunnels their data with us, with this setup in their Cisco VPN.

![image](https://user-images.githubusercontent.com/2943127/144542824-05ac4282-cd22-48d1-9945-20c0a98a9bd7.png)

This is our final `ipsec.conf `. Is this ok? missing something?

```
# ipsec.conf - strongSwan IPsec configuration file

config setup
    # cachecrls=yes
    # charonstart=no
    # strictcrlpolicy=yes
    # uniqueids=no
    # charondebug="dmn 0, mgr 0, ike 1, chd 0, job 0, cfg 1, knl 1, net 1, enc 0, lib 0"

conn %default
    ikelifetime=3h
    lifetime=5m
    margintime=1m
    keyingtries=30
    authby=psk
    keyexchange=ike
    mobike=no
    ike=3des-md5-modp1024!
    esp=3des-md5-modp1024-noesn!

conn CLARO
    left=192.168.10.20
    leftsubnet=192.168.10.0/24
    leftid=x.x.x.x
    leftfirewall=yes
    right=z.z.z.z
    rightsubnet=172.16.0.0/20
    rightid=z.z.z.z
    compress=no
    auto=add
```

Thanks!

**Reply by tobiasbrunner** [reply]
> Can I asked if my configuration is optimal, specially for `ikelifetime `and `lifetime`?

That completely depends on your requirements ([ExpiryRekey](https://wiki.strongswan.org/projects/strongswan/wiki/ExpiryRekey) has more information regarding rekeying and those settings), `lifetime` is quite low, though (the default is 1h) and `margintime` maybe too (the default [retransmission timeout](https://wiki.strongswan.org/projects/strongswan/wiki/Retransmission) is longer than 1 minute).


> Is this ok?

Again, depends on your requirements (e.g. if the connection should be triggered by traffic, instead of manually, in which case `auto=route` would be useful, DPD could also be, e.g. with `dpdaction=clear` with `auto=route` or  `dpdaction=restart` with `auto=add`). But note that you're using a legacy protocol (IKEv1) and very weak cryptographic algorithms (see [SecurityRecommendations](https://wiki.strongswan.org/projects/strongswan/wiki/SecurityRecommendations)).

**Reply by NormandoHall** [reply]
Tobias thanks for you guidance, it is very useful!

> But note that you're using a legacy protocol (IKEv1) and very weak cryptographic algorithms

Yes, this is the requirements for the peer, maybe has old Cisco VPN. Anyway there is no strict, because we are using their SIM cards to connect to their 4G network.

Regards!

PD: I have configured `ikelifetime=3h`, `lifetime=1h` and `margintime=9m`. This is the log received

```
2021-12-03 13:42:05 07[ENC] <CLARO|1> generating INFORMATIONAL_V1 request 2413977442 [ HASH N(DPD) ]
2021-12-03 13:42:05 08[ENC] <CLARO|1> parsed INFORMATIONAL_V1 request 2323836473 [ HASH N(DPD_ACK) ]
2021-12-03 13:42:25 10[IKE] <CLARO|1> sending keep alive to xx.xx.xx.xx[4500]
2021-12-03 13:42:31 11[ENC] <CLARO|1> parsed QUICK_MODE request 3521200592 [ HASH SA No KE ID ID ]
2021-12-03 13:42:31 11[CFG] <CLARO|1> selected proposal: ESP:3DES_CBC/HMAC_MD5_96/MODP_1024/NO_EXT_SEQ
2021-12-03 13:42:31 11[IKE] <CLARO|1> received 300s lifetime, configured 3600s
2021-12-03 13:42:31 11[IKE] <CLARO|1> received 4608000000 lifebytes, configured 0
2021-12-03 13:42:31 11[IKE] <CLARO|1> detected rekeying of CHILD_SA CLARO{2}
2021-12-03 13:42:31 11[ENC] <CLARO|1> generating QUICK_MODE response 3521200592 [ HASH SA No KE ID ID ]
2021-12-03 13:42:32 12[ENC] <CLARO|1> parsed QUICK_MODE request 3521200592 [ HASH ]
2021-12-03 13:42:32 12[IKE] <CLARO|1> CHILD_SA CLARO{3} established with SPIs cf30de5d_i de6bb068_o and TS 192.168.10.20/32 === 172.16.0.0/20
2021-12-03 13:42:55 06[IKE] <CLARO|1> sending keep alive to xx.xx.xx.xx[4500]
2021-12-03 13:43:05 08[IKE] <CLARO|1> sending DPD request
```

Specially this part:

```
2021-12-03 13:42:31 11[IKE] <CLARO|1> received 300s lifetime, configured 3600s
2021-12-03 13:42:31 11[IKE] <CLARO|1> received 4608000000 lifebytes, configured 0
```

Is this correct?
